technic_worldgen/oregen.lua

```diff
@@ -149,16 +152,30 @@ minetest.register_ore({
 local sulfur_buf = {}
 local sulfur_noise
 
+local has_mcl = minetest.get_modpath("mcl_core")
+
 minetest.register_on_generated(function(minp, maxp)
 	local vm, emin, emax = minetest.get_mapgen_object("voxelmanip")
 	local a = VoxelArea:new({MinEdge=emin, MaxEdge=emax})
 	vm:get_data(sulfur_buf)
 	local pr = PseudoRandom(17 * minp.x + 42 * minp.y + 101 * minp.z)
 	sulfur_noise = sulfur_noise or minetest.get_perlin(9876, 3, 0.5, 100)
 
-	local c_lava = minetest.get_content_id("default:lava_source")
-	local c_lava_flowing = minetest.get_content_id("default:lava_flowing")
-	local c_stone = minetest.get_content_id("default:stone")
+	local lava = "default:lava_source"
+	local lava_flowing = "default:lava_flowing"
+	local stone = "default:stone"
+	if has_mcl then
+	   lava = "mcl_core:lava_source"
+	end
+	if has_mcl then
+	   lava_flowing = "mcl_core:lava_flowing"
+	end
+	if has_mcl then
+	   stone = "mcl_core:stone"
+	end
```

Comment:
This is more about personal preference and some ppl round here might not share it, might want to wait a moment or argue against it if you do not like ternary style.

Personally I do prefer ternary when it is simple enough to be actually readable and if it allows throwing away very simple `if`-blocks.
```suggestion
	local lava = has_mcl and "mcl_core:lava_source" or "default:lava_source"
	local lava_flowing = has_mcl and "mcl_core:lava_flowing" or "default:lava_flowing"
	local stone = has_mcl and "mcl_core:stone" or "default:stone"
```

Reply:
done